Improve/Add Enums for Player, Items, and Quest

engineer124 opened the pull request:
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---
<!-- Leave the text above intact. Add additional comments below. -->

This PR does a few small things. 

1) Abbreviates many of the enums: (NOTE: there's open discussion on this one, particularly on the enums for the masks, so please share your thoughts either here or in discord to get a better sense of peoples opinions)
  - Discussion starts here: https://discord.com/channels/688807550715560050/688851997352263820/876084329120432148
  - An alternative set of names for `PlayerActionParam` can be seen here: https://pastebin.com/hrXj2H9A

2) It fixes `gUpgradeShifts` and `gUpgradeCapacities`, in which the quiver upgrade data from `gUpgradeCapacities` was accidentally in `gUpgradeShifts`. The fix involved updated the `CUR_CAPACITY` macro, updating `variables.h`, and `variables.txt`. 

3) Fills out the following enums:
  - `UpgradeType`
  - `ItemId`

4) Introduces the following enums:
  - `EffectType`
  - `QuestItem`
  - `InventorySlot`
  - `PlayerActionParam`
  - `PlayerBoots` (an alternative name to this is welcome)

Thanks to Rylie for sharing the resources to figure out `itemID` and `PlayerActionParam`
Thanks to EllipticEllipsis for figuring out the `PlayerBoots` enum
Thanks to Krim via kz for the info on `QuestItem` enum

AngheloAlf commented on tools/disasm/disasm.py:
```diff
@@ -91,6 +91,8 @@ def proper_name(symbol, in_data=False, is_symbol=True):
     # hacks
     if symbol == 0x809C46F0: # ovl_En_Encount4 fake symbol at the very end of the data section
         return variables_ast[0x809C46DC][0] + " + 0x14"
+    elif symbol == 0x801EF66D: # z_message_nes constant-folding stray fairy array
+        return variables_ast[0x801EF670][0] + f" - 0x{0x801EF670 - 0x801EF66D:X}"
```
What's this? :o

engineer124 replied:
This is Thar0's solution to the fake bss that was in collision_check.c that was put in shawlucas' PR. It just so happened to be the solution I needed as well to fix my bss problems so I grabbed it from that PR

AngheloAlf commented on src/code/z_en_item00.c:
```diff
@@ -206,7 +206,7 @@ void EnItem00_Init(Actor* thisx, GlobalContext* globalCtx) {
             Item_Give(globalCtx, ITEM_RUPEE_PURPLE);
             break;
         case ITEM00_RUPEE_ORANGE:
-            Item_Give(globalCtx, ITEM_RUPEE_ORANGE);
+            Item_Give(globalCtx, ITEM_RUPEE_GOLD);
```
Is this right?

engineer124 replied:
I'm not entirely sure. 

There are both separate orange and gold rupees that give 200 rupees. Killing the Takkuri in Termina field drops an orange rupee that rewards you with 200 rupees. Yet giving the curiosity shop man Chateau/Big Poe/Gold dust and you get a gold rupee worth 200 as well.

Something similar is in OoT where there are 6 gold rupee chests throughout the world that give 200 rupees, but you can also kill the skull kid which will drop an orange rupee worth 200 rupees as well.

The motivation for choosing gold over orange was because of the similarities between OoT's and MMs gold/orange rupees, so I just copied OoT. The proper way that needs to be done in MM is to document to distinguish between the two. At the moment though, I don't know which would be which in the enum.

I'll also admit that I did not notice the `ORANGE` in `ITEM00_RUPEE_ORANGE` until this comment

EllipticEllipsis replied:
I think it's one or other in the game text, it just happens that the item00 model doesn't look much like the GI model. And there is one in OoT that's worth 500, that you can get from diving game, IIRC.

engineer124 replied:
Which is which? Is GI what you get in the chest/npc and item00 is the freestanding item? Or is the other way around?

EllipticEllipsis replied:
GI is the "hold over the head" one

hensldm replied:
Also worth noting, when you sell the Chateau/Big Poe/Gold dust, it is referred to as a "Huge Rupee" with no reference to color.

EllipticEllipsis replied:
Should be that, then. Especially since the higher value ones have inconsistent colours in other games.

isghj5 replied:
You mean `ITEM_RUPEE_HUGE`? No complaints

engineer124 replied:
Right now we have:
`ITEM00_RUPEE_ORANGE`
`ITEM_RUPEE_GOLD`
`GI_RUPEE_GOLD`
Would you suggest to make all three Huge?
`ITEM00_RUPEE_HUGE`
`ITEM_RUPEE_HUGE`
`GI_RUPEE_HUGE`

AngheloAlf replied:
Yeah, as long as it is consistent I'm good with it.

AngheloAlf commented on src/code/z_collision_check.c:
```diff
@@ -1,4 +1,5 @@
 #include "global.h"
+#include "prevent_bss_reordering.h"
```
I just wanted to say the order in which you include this header affects bss reorder too. If you include it like this you may get different results:
```c
#include "prevent_bss_reordering.h"
#include "global.h"
```
I just wanted you to know this, you may need it in the future.

EllipticEllipsis replied:
Yes, the file itself says that it should go before any global header includes.

engineer124 replied:
That's really useful to know. Too useful actually, because moving it before `"global.h"` as suggested causes a NON_MATCH. However, moving it to the next file in the `spec` list, `code_800E8EA0` before `"global.h"` is alternative that matches OK and keeps it above `"global.h"`. Which would be preferred?

isghj5 commented on include/z64player.h:
```diff
@@ -7,50 +7,139 @@
+    /* 0x31 */ PLAYER_AP_DEED_OCEAN,
+    /* 0x33 */ PLAYER_AP_LETTER_MAMA = 0x33,
+    /* 0x36 */ PLAYER_AP_PENDANT_MEMORIES = 0x36,
+    /* 0x3A */ PLAYER_AP_MASK_TRUTH = 0x3A,
```
are we sure we know these missing values are never used? should we just include `PLAYER_AP_UNK32` or something incase?

engineer124 replied:
I started to fill them in, but then `UNK11` seemed ambiguous on whether it was hex or decimal so I just left them empty. I'm fine putting them in if that's something that seems strange only to me

engineer124 commented:
I have gone and removed the effects enums as they are also introduced in https://github.com/zeldaret/mm/pull/342, and are better reviewed in the context of the effects PR

rylieb reviewed (approved):
I like the current version of the names, with the possible exception of MASK_SCENTS instead of MASK_SCENT. I like trimming the S off for the apostrophe-S masks, but for an actual plural like Mask of Scents, I could see leaving it on. But honestly I'm not too bothered either way, so I'll approve this PR.

engineer124 commented on src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:
```diff
@@ -220,21 +220,22 @@ s32 func_80C10B0C(EnThiefbird* this, GlobalContext* globalCtx) {
-    s16 sp1E = 0;
+    s32 slotId = SLOT_BOTTLE_1;
+    s32 itemId1;
+    s16 itemId2 = ITEM_OCARINA;
```
I'm not sure a define like that should be local to this actor as setting it to 0 likely has significance in `func_801149A0` instead of within this file. I think I'd prefer to replace `ITEM_OCARINA` to 0 and wait until `func_801149A0` is decomped before overlapping both `ITEM_OCARINA` and `ITEM_NONE`, thoughts?

hensldm replied:
I'm fine with replacing it with a 0, mainly just removing the `ITEM_OCARINA`. However I think a local define works since `func_801149A0` never gets called with `itemId2=0`, since `isItemFound` stays false.
